examples/mysql/mysql_status.rb

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env ruby
+
+# Ensure your instrument_server process runs with these environment variables set,
+# or replace with your own.
+
+MYSQL_HOST              = ENV["MYSQL_HOST"]
+MYSQL_PORT              = ENV["MYSQL_PORT"] || 3306
+MYSQL_USER              = ENV["MYSQL_USER"]
+MYSQL_DEFAULTS_FILE     = ENV["MYSQL_DEFAULTS_FILE"]
+MYSQL_PASSWORD          = ENV["MYSQL_PASSWORD"]
+
+RATE_METRICS_TO_INSPECT = %w{Queries Bytes_sent Bytes_received Connections Slow_queries}
+CANARY_METRIC           = "Queries"
+
+require 'shellwords'
+
+env  = {}
+args = "-N -B -e 'SHOW GLOBAL STATUS' --user %s --host %s --port %s" % [MYSQL_USER, MYSQL_HOST, MYSQL_PORT].map { |arg| Shellwords.escape(arg) }
+
+if MYSQL_DEFAULTS_FILE.to_s.size > 0
+  args = "--defaults-file=%s %s" % [Shellwords.escape(MYSQL_DEFAULTS_FILE), args]
+else
+  env = { "MYSQL_PWD" => MYSQL_PASSWORD }
+end
+
+cmd = ["mysql", args].join(" ")
+
+stdout_r, stdout_w = IO.pipe
+
+pid = Process.spawn(env, cmd, :out => stdout_w, :err => STDERR)
+
+pid, exit_status = Process.wait2(pid)
+
+stdout_w.close
+
+previous_run_time = ARGV[0].to_i
+current_time      = Time.now.to_i
+run_interval      = (current_time - previous_run_time).to_f
```

Comment:
so part of what comes back from mysql is an uptime variable.  You can use that for two things - 1) the canary metric, if uptime is less than last see uptime, and 2) for more correct calculation of time between current info and previous info